### ISSUE TITLE
Dice loss operating on logits rather than predictions

### ISSUE THREAD

**Issue by Zach-ER** [jakeret/tf_unet#15]
In calculating the dice loss, we should be taking the softmax of the logits before using it to calculate the loss function.

**Comment by jakeret** [comment]
You could be right. I saw that you started to investigate a little bit. 
Let me know what you find

**Comment by shiba24** [comment]
What if we like to compute dice loss for multi classification?

**Comment by jakeret** [comment]
@shiba24 I tried that using the Circle and Rectancle DataProvider in the `unet` package and I seemed to work alright. 
Could you be a bit more specific?

**Comment by jakeret** [comment]
I just merged a branch that should tackle this issue into master. I'm closing this, let me know if you encounter any problems
